### PR TITLE
feature: integrate firestore storage rules

### PR DESCRIPTION
**Pull request by mkue**

### CODE REVIEW

**Comment by mkue** [comment]
Oh wow, this took me forever🙈 Renaming the project from `demo-social-income-prod` to `social-income-prod` lead to some weird side-effects with the tests because the document references we use in some firestore collections, e.g. recipient.organisation, pointed to documents in the emulated `demo-social-income-prod` firestore. Unfortunately, you never see the full document path with the project id included, so it wasn't obvious why some document queries in the tests just didn't work...

**Comment by andrashee** [comment]
Very nice @mkue, I will have a look at it tomorrow!

File: .github/workflows/firestore-rules.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v2
+
+      - name: Install firebase-tools
+        run: npm -g install firebase-tools
+
+      - name: Deploy to Firebase
+        if: github.ref == 'refs/heads/main'
+        run: firebase deploy --only storage --token ${{ secrets.FIREBASE_TOKEN }} --project social-income-prod
```
Comment (andrashee) [inline .github/workflows/firestore-rules.yml:33]: is this correct or should it be `--only firestore:rules`?

File: firebase.json [hunk 2]
```diff
@@ -29,7 +29,7 @@
 		"runtime": "nodejs16"
 	},
 	"storage": {
-		"rules": "storage.rules"
+		"rules": "firestore.rules"
```
Comment (andrashee) [inline firebase.json:32]: Is this change on purpose? Here we are talking about storage rules not firestore no?

File: functions/package.json [hunk 3]
```diff
@@ -9,9 +9,8 @@
 		"build": "tsc",
 		"build:watch": "tsc --watch",
 		"serve": "npm run build:watch",
-		"serve:watch": "npm run build -- --watch & firebase emulators:start --project social-income-prod --config ../firebase.json --import ../seed",
-		"test": "firebase emulators:exec --only functions --only firestore --project social-income-prod --config ../firebase.json 'npm run test:app'",
-		"test:app": "jest --roots  src/"
+		"test": "firebase emulators:exec --only functions --only firestore --project social-income-prod --config ../firebase.json --import ../seed 'npm run test:app'",
```
Comment (andrashee) [inline functions/package.json:12]: Why did you delete the serve:watch? I find it useful to work with.

File: recipients_app/.run/prod.run.xml [hunk 4]
```diff
@@ -1,7 +0,0 @@
-<component name="ProjectRunConfigurationManager">
```
Comment (andrashee) [inline recipients_app/.run/prod.run.xml:1]: Do you delete this on purpose?

File: recipients_app/.run/dev.run.xml [hunk 5]
```diff
@@ -1,7 +0,0 @@
-<component name="ProjectRunConfigurationManager">
```
Comment (andrashee) [inline recipients_app/.run/dev.run.xml:1]: Do you delete this on purpose?

File: storage.rules [hunk 6]
```diff
@@ -1,8 +0,0 @@
-rules_version = '2';
```
Comment (andrashee) [inline storage.rules:1]: Why did you delete this one?

**Review by andrashee — COMMENTED** [review]
Very nice 👍 

Is the current file up-to-date with the in firebase?

File: .github/workflows/firestore-rules.yml (hunk 1 above)
Comment (mkue) [reply .github/workflows/firestore-rules.yml:33]: Oh wow, you're absolutely right. This is completely wrong!

I changed it, we deploy both rules files now.

File: firebase.json (hunk 2 above)
Comment (mkue) [reply firebase.json:32]: Yes, you're right. I totally mixed this up.

File: recipients_app/.run/prod.run.xml (hunk 4 above)
Comment (mkue) [reply recipients_app/.run/prod.run.xml:1]: Hmm, I think I removed some run configurations in WebStorm that were suddenly there. I've never shared IDE run configurations in a repository and I don't think we should. What do you think?

File: recipients_app/.run/dev.run.xml (hunk 5 above)
Comment (mkue) [reply recipients_app/.run/dev.run.xml:1]: Same as above

File: storage.rules (hunk 6 above)
Comment (mkue) [reply storage.rules:1]: Because I made a mess😅

File: functions/package.json (hunk 3 above)
Comment (mkue) [reply functions/package.json:12]: Oh sorry, this is a mistake. I thought that it can't be executed, which is totally wrong.

**Comment by mkue** [comment]
@andrashee, please quickly take a look again.

> Is the current file up-to-date with the in firebase?
Yes it is👍🏼

**Review by andrashee — APPROVED** [review]
That's awesome! I find it a massive step to have those rules manageable and testable. 🥳

**Comment by ssandino** [comment]
🎉🎉🎉
